src/common/utils/client-ip.util.ts

```
@@ -16,25 +16,74 @@ export function normalizeIpForForwarding(ip: string | undefined): string | undef
 }
 
 /**
- * Get the real client IP from the request.
- * Use when the app is behind a trusted proxy (e.g. AWS ALB) that sets X-Forwarded-For.
- * Keycloak will log this IP when the backend forwards it on the token request.
- * Safe to call with null/undefined or non-Express request; returns undefined instead of throwing.
+ * True if the IP is private/internal or loopback. Used to decide when it's safe
+ * to trust X-Forwarded-For (only when the direct connection is from our infra).
+ */
+function isPrivateOrLoopback(ip: string | undefined): boolean {
+  if (!ip || typeof ip !== 'string') return false;
```

ESLint

[error] Replace `'string'` with `"string"`

(prettier/prettier)